components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonBuilder.java

```diff
@@ -56,4 +69,66 @@ public void setAllowStAX(boolean allowStAX) {
     protected XmlSourceHandlerFactoryImpl createXmlSourceHandlerFactoryImpl() {
         return new SaxonXmlSourceHandlerFactoryImpl();
     }
+
+    @Override
+    protected Templates createTemplates(TransformerFactory factory, Source source) throws TransformerConfigurationException {
+        final Templates templates = super.createTemplates(factory, source);
+        if (templates instanceof TemplatesImpl templatesImpl && getXsltMessageLogger() != null) {
```

Comment:
Sorry again one of those java 17 only, if this feature should be backported to 3.x

Reply:
Fixed.